tests/unit/test_restore_function.py

```diff
@@ -1,54 +1,62 @@
 import os
-import unittest
-from mock import patch, Mock
 import sys
-sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__),'../../src/restore')))
-import mock_import
-import restore_function
+import unittest
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__),'../../src')))
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__),'../../src/common/python')))
+import custom_exceptions
+from mock import patch
+from restore import restore_function
 
+os.environ["Region"] = "us-west-2"
+
+@patch("restore.restore_function.boto3.client")
 class TestResourceProvider(unittest.TestCase):
-    def test_restore_success(self):
-        os.environ["Region"] = "us-west-2"
-        factory_patch = patch('restore_function.boto3.client')
-        mock_factory_boto_client = factory_patch.start()
-        mock_response = Mock(name='response')
-        mock_factory_boto_client.return_value = mock_response
-        mock_response.return_value = {"taskname": "Restore", "identifier": "database-1"}
-        mock_factory_boto_client.describe_db_instances.return_value = mock_response
-        mock_response.describe_db_instances.return_value = {"DBInstances": [{"DBInstanceIdentifier": "database-1-temp","VpcSecurityGroups": [{"VpcSecurityGroupId": "abc"}],"DBSubnetGroup": {"DBSubnetGroupName": "xyz"}}]}
-        event = create_event()
-        data = restore_function.lambda_restore_dbinstance(event, {})
+    def setUp(self):
+        self.event = {"identifier": "database-1-temp"}
+        self.restored_instance_id = "database-1"
+        self.mocked_rate_exceeded_exception = custom_exceptions.RateExceededException("Identifier:database-1 \nthrottling error: Rate exceeded")
+        self.mocked_instance_not_found_exception = custom_exceptions.InstanceRestoreException("Identifier:database-1 \nDBInstanceNotFound")
+        self.mocked_duplicate_instance_exception = custom_exceptions.InstanceRestoreException("Identifier:database-1 \nDBInstanceAlreadyExistsFault")
+        self.mocked_describe_db_instances = {
```

Comment:
Can we move sample responses to a prepare utility method instead of hardcoding them here.